src/skill.cpp

```diff
@@ -135,6 +135,23 @@ bool Skill::is_contextual_skill() const
     return _tags.count( "contextual_skill" ) > 0;
 }
 
+Skill::skill_type Skill::get_skill_type() const
+{
+    if( _tags.count( "melee_skill" ) > 0 ) {
```

Comment:
It would be better and easier to use if the skill type works like an item category: have a separate JSON member that defines the category and have another JSON types that defines name of skill categories and maybe sort rank.

Currently the game does not handle skills with multiple types well (e.g. if I give my new skill the "ranged_skill" *and* the "social_skill" tag, it will be listed as "ranged skill"). Also: mods can't define new skill types.